.github/workflows/generate-json.yml

```
@@ -0,0 +1,36 @@
+# Regenerates JSON schema whenever a PR is merged to main 
+name: Generate JSON
+
+on:
+  push:
+    branches:
+      - main
+      # Testing only
+      # - docs/json-schema
```

```suggestion
  pull_request:
    paths-ignore:
      - 'README.md'
```

How about running this only on pull requests so that we can review the commit before merging to main?